ci-operator/step-registry/openshift/e2e/libvirt/test/openshift-e2e-libvirt-test-commands.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export PATH=/usr/libexec/origin:$PATH
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+# Initial check
+if [[ "${CLUSTER_TYPE}" != "libvirt-ppc64le" ]] && [[ "${CLUSTER_TYPE}" != "libvirt-s390x" ]] ; then
+    echo "Unsupported cluster type '${CLUSTER_TYPE}'"
+    exit 0
+fi
+
+function upgrade() {
+    set -x
+    openshift-tests run-upgrade all \
+        --to-image "${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}" \
+        --options "${TEST_UPGRADE_OPTIONS-}" \
+        --provider "${TEST_PROVIDER}" \
+        -o "${ARTIFACT_DIR}/e2e.log" \
+        --junit-dir "${ARTIFACT_DIR}/junit"
+    set +x
+}
+
+function suite() {
+    if [[ -f "${SHARED_DIR}/excluded_tests" ]]; then
+
+        cat > ${SHARED_DIR}/invert_excluded.py <<EOSCRIPT
+#!/usr/libexec/platform-python
+import sys
+all_tests = set()
+excluded_tests = set()
+for l in sys.stdin.readlines():
+  all_tests.add(l.strip())
+with open(sys.argv[1], "r") as f:
+  for l in f.readlines():
+    excluded_tests.add(l.strip())
+test_suite = all_tests - excluded_tests
+for t in test_suite:
+  print(t)
+EOSCRIPT
+chmod +x ${SHARED_DIR}/invert_excluded.py
+
+
+openshift-tests run openshift/conformance/parallel --dry-run | ${SHARED_DIR}/invert_excluded.py ${SHARED_DIR}/excluded_tests > ${SHARED_DIR}/tests
```

Comment:
This only makes sense for `conformance/parallel` tests. If you use another suite as parameter to `run` but give it a list of `conformance/parallel` tests, won't it run the wrong set of tests? At least that's what happened when I tried it.

I'd even say that generating the list of files to run should be part of the `conf` step.

Reply:
Make you could make this line more generic/correct by using `"${TEST_SUITE}"` instead of hard-coding `openshift/conformance/parallel`